src/sc/contentrules/localrole/interfaces.py

```diff
@@ -1,26 +1,64 @@
 # -*- coding: utf-8 -*-
+from plone.autoform.interfaces import IFormFieldProvider
+from plone.supermodel import model
 from sc.contentrules.localrole import MessageFactory as _
+from sc.contentrules.localrole.config import DEFAULT_ENABLED_GROUPS
+from zope import schema
 from zope.interface import Interface
-from zope.schema import Choice
-from zope.schema import Set
-from zope.schema import TextLine
+from zope.interface import provider
 
 
 class ILocalRoleAction(Interface):
     """An action used to apply a local role to an object
     """
 
-    principal = TextLine(title=_(u"Username / Group name"),
-                         description=_(u"Please inform the username or "
-                                       u"groupname that will be used by "
-                                       u"this action.  Use  ${title} in "
-                                       u"this field to use the content "
-                                       u"title as the value for this field."),
-                         required=True)
-
-    roles = Set(title=_(u"Roles"),
-                description=_(u"Roles to be assigned to user/group "
-                              u"in the object."),
-                required=True,
-                value_type=Choice(vocabulary='plone.app.vocabularies.Roles')
-                )
+    principal = schema.TextLine(
+        title=_(u"Username / Group name"),
+        description=_(u"Please inform the username or "
+                      u"groupname that will be used by "
+                      u"this action.  Use  ${title} in "
+                      u"this field to use the content "
+                      u"title as the value for this field."
+                      u"  Use  ${localrole_group} in "
+                      u"this field to use the behavior "
+                      u"group as the value for this field."),
+                      required=True
+        )
+
+    roles = schema.Set(
+        title=_(u"Roles"),
+        description=_(
+            u"Roles to be assigned to user/group "
+            u"in the object."),
+        required=True,
+        value_type=schema.Choice(vocabulary='plone.app.vocabularies.Roles')
+    )
+
+
+@provider(IFormFieldProvider)
+class IPrincipleFromContext(model.Schema):
+
+    """Behavior interface to pick principle field from context item.."""
+
+    model.fieldset('localrole', label=_(u'Local Role'), fields=['localrole_group'])
+    localrole_group = schema.Choice(
+        title=_(u'Group'),
+        vocabulary=u'sc.contentrules.localrole.groups',
+        required=False,
```

Comment:
This field can be required if the use case is interesting
```suggestion
        required=True,
```